src/Task.h

```diff
@@ -790,6 +790,26 @@ class Task {
   void xptrace(int request, remote_ptr<void> addr, void* data);
 
   /**
+   * If the given memory region is mapped into the local address, space, invoke
+   * the callback with a pointer to the local equivalent of addr and the total
+   * number of bytes available through this mapping
+   */
+  template <typename callbackT>
+  ssize_t with_local_mapping(remote_ptr<void> addr, size_t size,
+                             callbackT callback) {
+    if (vm()->has_mapping(addr)) {
+      const AddressSpace::Mapping& map = vm()->mapping_of(addr);
+      if (map.local_addr != nullptr) {
+        size_t nbytes =
+            std::min(size, static_cast<uintptr_t>(map.map.end() - addr));
+        size_t offset = addr - map.map.start();
+        return callback(static_cast<uint8_t*>(map.local_addr) + offset, nbytes);
+      }
```

Comment:
The reason this was in Task.h was because both RecordTask and Task need it. I'll look into making it just return the pointer though, in which case, I can move it out of Task.h.